### ISSUE TITLE
BUG: Logit margeff summary TypeError

### ISSUE THREAD

**Issue by TomAugspurger** [statsmodels/statsmodels#2817]
Really the only example from my notebook cleanup that looks like a real bug.

``` python
import numpy as np
import statsmodels.api as sm

spector_data = sm.datasets.spector.load()
spector_data.exog = sm.add_constant(spector_data.exog, prepend=False)

logit_mod = sm.Logit(spector_data.endog, spector_data.exog)
logit_res = logit_mod.fit(disp=0)

margeff = logit_res.get_margeff()
print(margeff.summary())
```

raises

``` python
TypeError                                 Traceback (most recent call last)
<ipython-input-1-210bf23ad669> in <module>()
      9 
     10 margeff = logit_res.get_margeff()
---> 11 print(margeff.summary())

/Users/tom.augspurger/Envs/dev/lib/python3.4/site-packages/statsmodels/statsmodels/discrete/discrete_margins.py in summary(self, alpha)
    520         _, const_idx = _get_const_index(model.exog)
    521         if const_idx is not None:
--> 522             exog_names.pop(const_idx)
    523 
    524         J = int(getattr(model, "J", 1))

TypeError: only integer scalar arrays can be converted to a scalar index
```

**Comment by josef-pkt** [comment]
I don't really understand this `_get_const_index(model.exog)` just uses np.where

However, this code is outdated since const_idx is now attache to the model or model.data

Same code is in GEE (copied and adjusted from discrete)

no smoketests for `summary` method in test suite. 
discrete margins are only tested on Logit, AFAICS

(I haven't installed numpy 1.11.0.bxxx yet because I either don't remember how to set up a virtualenv, or the python built in virtualenv in 3.4 doesn't work properly, or it's because pip hates me back)

**Comment by josef-pkt** [comment]
the problem is that const_idx is a 1d array with one element, which cannot be used for indexing or popping a python list.

The newer code in base data squeezes: `np.where(...)[0].squeeze()`
AFAIK, the model const_idx is used quite often and should have proper functional unit test coverage

**Comment by theaverageguy** [comment]
@kshedden Fixed?

**Comment by kshedden** [comment]
I was not planning to fix this in Logit, I have not worked on that code at
all.

I have a pending PR (#2833) that fixes the same bug in GEE.  There are a
few more things that I need to deal with there before it can be merged
though.

**Comment by josef-pkt** [comment]
looks like we missed this one, I had thought it was incorporated somewhere

**Comment by josef-pkt** [comment]
The error was fixed in #3037, example runs fine with master
it still uses the outdated pattern with `_get_const_index(model.exog)`, not urgent

**Comment by josef-pkt** [comment]
I opened #3719 for the refactoring issue, so I can close this as fixed.
